modules/mobile/app/models/mobile/v0/authorized_services.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    class AuthorizedServices
+      def initialize(user)
+        @user = user
+      end
+
+      def to_a
+        authorized_services_map.select { |_, v| v == true }.keys
+      end
+
+      def available_services
+        authorized_services_map.keys
+      end
+
+      private
+
+      def authorized_services_map
+        {
+          appeals: access?(appeals: :access?),
+          appointments: true,
+          claims: flagged_access?(:mobile_lighthouse_claims, { lighthouse: :access? }, { evss: :access? }),
+          directDepositBenefits: flagged_access?(:mobile_lighthouse_direct_deposit, { lighthouse: :mobile_access? },
+                                                 { evss: :access?, ppiu: :access? }),
+          directDepositBenefitsUpdate: direct_deposit_update_access?,
+          disabilityRating: access?(evss: :access?),
+          lettersAndDocuments: flagged_access?(:mobile_lighthouse_letters, { lighthouse: :access? },
+                                               { evss: :access? }),
+          militaryServiceHistory: access?(vet360: :military_access?),
+          paymentHistory: access?(bgs: :access?),
+          userProfileUpdate: access?(vet360: :access?),
+          secureMessaging: access?(mhv_messaging: :access?),
+          scheduleAppointments: access?(schedule_appointment: :access?),
+          prescriptions: access?(mhv_prescriptions: :access?),
+          preferredName: access?(demographics: :access?) && access?(mpi: :queryable?),
+          genderIdentity: access?(demographics: :access_update?) && access?(mpi: :queryable?)
```

Comment:
Updated preferredName and genderIdentity to match the controllers. The current definitions were not correct because the code auth services code was too rigid to allow them to be correct.